src/tests/AsyncTests.fs

```diff
@@ -200,18 +200,22 @@ open Fable.Core
 
 [<Test>]
 let ``Interaction between Async and Promise works``() =
-    let res = ref false
-    async { res := true }
-    #if FABLE_COMPILER
-    |> Async.StartAsPromise
-    |> Async.AwaitPromise
-    #else
-    |> Async.StartAsTask
-    |> Async.AwaitTask
-    #endif
-    |> Async.StartImmediate
-    equal true !res
-    
+    async {
+        let res = ref false
+        async { res := true }
+        #if FABLE_COMPILER
+        |> Async.StartAsPromise
+        |> Async.AwaitPromise
+        #else
+        |> Async.StartAsTask
+        |> Async.AwaitTask
+        #endif
+        |> Async.StartImmediate
+        // some wait is needed on Linux for some reason
```

Comment:
The semantics of `Async.Start/StartImmediate` is translated very loosely in Fable (actually both methods have the same effect). As in JS you cannot define _true_ async methods to run in a separate thread (unless you use web workers), it's difficult to know if an async workflow will actually run in a separate thread. This only depends if native APIs (HTTP requests, timers, events...) are used.

The purpose of this test is just to check that converting async to promise back and forth doesn't throw any error.

> Note that `Async.RunSynchronously` is forbidden except for unit tests.